plugin/template.go

```diff
@@ -203,8 +225,16 @@ func (g *goFileGenerator) Generate(filename, tmpl string, data interface{}) ([]b
 		return nil, fmt.Errorf("failed to parse template %q: %v", filename, err)
 	}
 
+	for idx, additionalT := range g.additionalTmpls {
+		t, err = t.New("template_" + strconv.Itoa(idx)).Parse(additionalT)
```

Comment:
We don't have a specific name with the current API. You just pass in a template string. I could add a second parameter for the template name when you add the template so that the error messages are clearer.

> If there's an issue with duplicate file base names, what are your thoughts on banning those until there's sufficient demand for that feature?

Not sure I follow this. You don't provide file names, you're providing template strings directly. There can't be a conflict, especially since we don't currently allow a name.

>With template calls outside the loop, I think we can drop t.Lookup in favor of the result from ParseFiles.

OK, I can simplify this.

Reply:
You solved the issue I was thinking about. Since you specify that the names are provided directly then there shouldn't really be an issue with a conflict.

I was thinking about the documentation on ParseFile "When parsing multiple files with the same name in different directories, the last one mentioned will be the one that results." when I made this comment about name collisions thinking that filepaths were the next way of naming the provided template.